plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -182,14 +190,44 @@ export class TaskWorker {
     });
   }
 
+  protected truncateParameters(parameters: JsonObject) {
+    const auditMaxLength =
+      this.config?.getOptionalNumber('scaffolder.auditor.maxLength') ?? 256;
+
+    if (auditMaxLength === -1) {
+      this.logger?.debug(
+        `scaffolder.auditor.maxLength manually disabled via configuration, no task parameter length limit set.`,
+      );
+      return parameters;
+    }
+
+    const truncatedParameters: JsonObject = {};
+
+    for (const key in parameters) {
```

Comment:
Afaik `parameters` can have nested objects, so this would need to be recursive

Reply:
Yup, have implemented a little more complexed function to loop the parameters, also added some test cases.